server/apps/immich/src/api-v1/system-config/dto/system-config-update-oauth.dto.ts

```diff
@@ -0,0 +1,36 @@
+import { IsBoolean, IsNotEmpty, IsOptional, IsString, ValidateIf } from 'class-validator';
+
+const isEnabled = (config: SystemConfigUpdateOAuthDto) => config.enabled;
+
+export class SystemConfigUpdateOAuthDto {
+  @IsOptional()
+  @IsBoolean()
+  enabled!: boolean;
+
+  @ValidateIf(isEnabled)
+  @IsNotEmpty()
+  @IsString()
+  issuerUrl!: string;
+
+  @ValidateIf(isEnabled)
+  @IsNotEmpty()
+  @IsString()
+  clientId?: string;
```

Comment:
Oh wait, no issuer url, client id, and client secret should all have ? because they are optional unless enabled is true. If you are turning oauth off, you should not be required to pass a client id, secret, or issuer url to do so.